include/revng/Model/Pass/RegisterModelPass.h

```diff
@@ -16,27 +17,50 @@ class RegisterModelPass {
   using ModelPass = std::function<void(TupleTree<model::Binary> &)>;
 
 private:
-  static llvm::ManagedStatic<std::map<std::string, ModelPass>> Registry;
+  struct Registered {
+    std::string Name;
+    std::string Description;
+    ModelPass Pass;
+  };
+  struct TransparentNameComparator {
+    using is_transparent = std::true_type;
+
+    template<typename Type>
+    llvm::StringRef get(const Type &Value) const {
+      return llvm::StringRef(Value);
+    }
+    llvm::StringRef get(const Registered &Value) const {
+      return llvm::StringRef(Value.Name);
+    }
+    template<typename LHS, typename RHS>
+    bool operator()(const LHS &Left, const RHS &Right) const {
+      return get(Left) < get(Right);
+    }
+  };
+  using RegistryType = std::set<Registered, TransparentNameComparator>;
+  static llvm::ManagedStatic<RegistryType> Registry;
 
 public:
-  RegisterModelPass(const llvm::Twine &Name, ModelPass Pass) {
-    bool Inserted = Registry->emplace(Name.str(), Pass).second;
-    revng_assert(Inserted);
+  RegisterModelPass(const llvm::Twine &Name,
+                    const llvm::Twine &Description,
+                    ModelPass Pass) {
+    Registered RegisteredPass{ Name.str(), Description.str(), Pass };
+    auto [_, Success] = Registry->emplace(std::move(RegisteredPass));
+    revng_assert(Success);
   }
 
 public:
   static const ModelPass *get(llvm::StringRef Name) {
-    auto It = Registry->find(Name.str());
-    if (It == Registry->end())
+    if (auto It = Registry->find(Name); It == Registry->end())
```

Comment:
I'm still not used to this syntax, but it's definitely something we want to adopt more!